_posts/loadbalancers/2024-03-08-create-a-ssl-loadbalancer.md

```diff
@@ -0,0 +1,361 @@
+---
+layout: page
+tags: [Loadbalancers]
+page_title: Create a SSL loadbalancer
+---
+
+This tutorial guides you through the process of creating a load balancer with SSL encryption. 
+The article assumes there is an internal network present with working http web servers listening on 
+port 80. When you still need to create webservers and an internal network, please use the first part 
+of article [Create a loadbalancer with webservers]({{ '/articles/create-a-loadbalancer-with-webservers' | relative_url }})
+
+
+---
+
+## SSL certificate
+Before creating the load balancer, we need to store our SSL certificaten in OpenStack. We are using 
+Keymanager to do so. To read more about Keymanaer, refer to the article 
+[Introduction to Keymanager]({{ '/articles/introduction-to-keymanager' | relative_url }}).
+Currently it is not possible to upload the certificate through Horizon so we will be using the CLI.
+There are multiple options to upload the certificate to barbican. Our advise would be to use the
+container approach 
+([Uploading the SSL certificate to keymanager in a container](#uploading-the-ssl-certificate-to-keymanager-in-a-container))
+. The easier option is the combined, although slightly less secure, approach
+which has the benefit of being selectable in horizon after storing 
+([Uploading the SSL certificate to keymanager as single file](#uploading-the-ssl-certificate-to-keymanager-as-single-file))
+.
+
+### Uploading the SSL certificate to keymanager in a container
+
+The prefered way of storing a certificate in barbican for Octavia is to use seperate secrets to
+store the server certificate, private key, intermediate certificates and the passhprase. After
+storing the secrets, we can combine them using a certificate container. The downside of this
+approach however is secret containers are not yet supported in Horizon to use for Octavia.
+
+**Prerequisites**
+ - All certificates files are stored on the OpenStack CLI server. We need the following files:
+   - certificate.pem (the certificate file for the load balancer)
+   - private.key (the private key for the load balancer, password protected)
+   - intermediate.pem (intermediate certificates in proper order of your SSL supplier)
+ - Passphrase to decrypt the private key
+ - openssl tooling installed on the OpenStack CLI server
+
+
+**Step 1**  
+First make sure you have setup the OpenStack CLI and that you are able to
+execute commands using the `openstack` command. For more information please
+refer to the
+[Using the OpenStack CLI article](
+    {{ '/articles/using-the-cli-linux' | relative_url }}
+).
+
+**Step 2**
+
+Store the certificate in barbican
+```bash
+certificate=certificate.pem
```

Comment:
Please add some comments and/or some whitespace so the article is more readable 

An example of this you can find at the bottom of the [Resize Volume](https://cloudtutorials.eu/articles/resize-volume) article (Check the last code block)